### PR TITLE
ROAD-502: Release 4.0.0-rc

### PR DESCRIPTION
**Pull request by parkhomenko**
Release candidate version for C# driver

### CODE REVIEW

**Comment by faunaee** [comment]
Note that the C# driver is not appear to provide implementations for the following functions:

* All
* Any
* IsArray
* IsBoolean
* IsBytes
* IsCollection
* IsCredentials
* IsDatabase
* IsDate
* IsDoc
* IsDouble
* IsFunction
* IsIndex
* IsInteger
* IsKey
* IsLambda
* IsNull
* IsNumber
* IsObject
* IsRef
* IsRole
* IsSet
* IsString
* IsTimestamp
* IsToken
* TimeAdd
* ToArray
* ToDouble
* ToInteger
* ToObject

These functions were all added to FQL prior to API v3, so it would be a major omission if they were (still) not implemented in a v4 driver.

**Comment by n400** [comment]
For Ewan’s note: https://faunadb.atlassian.net/browse/DRV-598

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 4.0.0-rc
+- Streaming support
+- 3rd party auth support
+- HTTP/2 support for .net standard 2.1 and above
```
Comment (n400) [inline CHANGELOG.md:4]: List all of the missing functions that got added?

**Review by n400 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (parkhomenko) [reply CHANGELOG.md:4]: added

File: FaunaDB.Client/FaunaDB.Client.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>net45;netstandard1.5;netstandard2.0;netstandard2.1</TargetFrameworks>
-    <Version>3.0.0</Version>
+    <Version>4.0.0-rc</Version>
     <PackageId>FaunaDB.Client</PackageId>
```
Comment (evbruno) [inline FaunaDB.Client/FaunaDB.Client.csproj:6]: dumb question here: is the `rc` the same purpose of "snapshots" for Maven repo (JVM) ? Can we override `rc` releases?

**Review by evbruno — APPROVED** [review]

File: FaunaDB.Client/FaunaDB.Client.csproj (hunk 2 above)
Comment (parkhomenko) [reply FaunaDB.Client/FaunaDB.Client.csproj:6]: I think they have different meanings. Maven's snapshot version is not considered a release version and can be updated, so a user who uses 3.0.0-snapshot, for example, can expect that there can be new changes within this same version.
Nuget's `rc` can't be changed in the same way, if we have new changes we need to publish another package, with, say, version 4.0.0-rc1, 4.0.0-rc2, and so on.

We can write anything after the "-", but conventionally we use alpha, beta, rc if we mean pre-release version. There should be a checkbox in visual studio to include pre-release version, so users can use them in their projects.
And they are also marked as pre-release on nuget.org:
https://www.nuget.org/packages/FaunaDB.Client/1.2.1-alpha

**Review by n400 — APPROVED** [review]

**Review by updatus — APPROVED** [review]
LGTM.

**Review by vadimLF — APPROVED** [review]
LGTM.

**Pull request merged**